src/Orchard.Hosting.Web/Routing/ModuleAreaRouteConstraintApplicationModelProvider.cs

```diff
@@ -31,7 +31,8 @@ public void OnProvidersExecuted(ApplicationModelProviderContext context)
                 var feature = _typeFeatureProvider.GetFeatureForDependency(controller.ControllerType.AsType());
                 if (feature != null)
                 {
-                    controller.RouteConstraints.Add(new AreaAttribute(feature.Descriptor.Id));
+                    var areaAttribute = new AreaAttribute(feature.Descriptor.Id);
+                    controller.RouteValues.Add(areaAttribute.RouteKey, areaAttribute.RouteValue);
```

Comment:
Interesting, I did this....

controller.RouteValues.Add("area", feature.Descriptor.Id);